### PR TITLE
Update locale-info.php

### PR DESCRIPTION
**Pull request by tpaksu**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Related fix for https://github.com/Automattic/woocommerce-payments/issues/2246.

This PR will add support for other currencies and locale specific settings for WooCommerce, which will also solve the need for locale-currency-country matching when the customer's location is detected via geolocation system, also will be used as a base for the other multi-currency related plugins, e.g. WooCommerce Payments.

It includes the new country and locale data while preserving the original content, with some additional formatting fixes, all obtained from data provided by the CLDR (http://cldr.unicode.org/).  

Existing keys still remain the same:

- **currency_code**: The 3-letter ISO-4217 currency code
- **currency_pos**: Whether the symbol is placed on left or right while formatting
- **thousand_sep**: The thousand separator used in the default locale for the country
- **decimal_sep**: The decimal separator used in the default locale 
- **num_decimals**: Number of decimal digits used for the currency
- **weight_unit**: Common unit of weight used in the country
- **dimension_unit**: Common unit of length used in the country

New keys added to the existing country data:

- **direction**: The text direction of the country's default locale
- **default_locale**: Default locale for the country
- **name**: Name of the currency
- **singular**: Singular form of the currency display
- **plural**: Plural form of the currency display
- **short_symbol**: Currency symbol
- **locales**: The locales used for the country

The new countries are:

| Country | Country | Country | Country | Country |
| --- | --- | --- | --- | --- |
| **AD**: Andorra | **AE**: United Arab Emirates | **AF**: Afghanistan | **AG**: Antigua & Barbuda | **AI**: Anguilla | **AL**: Albania |
| **AM**: Armenia | **AO**: Angola | **AR**: Argentina | **AS**: American Samoa | **AW**: Aruba |
| **AX**: Åland Islands | **AZ**: Azerbaijan | **BB**: Barbados | **BF**: Burkina Faso | **BG**: Bulgaria |
| **BH**: Bahrain | **BI**: Burundi | **BJ**: Benin | **BL**: St. Barthélemy | **BM**: Bermuda |
| **BN**: Brunei | **BO**: Bolivia | **BQ**: Caribbean Netherlands | **BS**: Bahamas | **BT**: Bhutan |
| **BW**: Botswana | **BY**: Belarus | **BZ**: Belize | **CC**: Cocos (Keeling) Islands | **CD**: Congo - Kinshasa |
| **CF**: Central African Republic | **CG**: Congo - Brazzaville | **CI**: Côte d’Ivoire | **CK**: Cook Islands | **CL**: Chile |
| **CM**: Cameroon | **CN**: China | **CO**: Colombia | **CR**: Costa Rica | **CU**: Cuba |
| **CV**: Cape Verde | **CW**: Curaçao | **CX**: Christmas Island | **CY**: Cyprus | **CZ**: Czechia |
| **DG**: Diego Garcia | **DJ**: Djibouti | **DM**: Dominica | **DO**: Dominican Republic | **DZ**: Algeria |
| **EA**: Ceuta & Melilla | **EC**: Ecuador | **EE**: Estonia | **EG**: Egypt | **EH**: Western Sahara |
| **ER**: Eritrea | **ET**: Ethiopia | **FJ**: Fiji | **FK**: Falkland Islands | **FM**: Micronesia |
| **FO**: Faroe Islands | **GA**: Gabon | **GD**: Grenada | **GE**: Georgia | **GF**: French Guiana |
| **GG**: Guernsey | **GH**: Ghana | **GI**: Gibraltar | **GL**: Greenland | **GM**: Gambia |
| **GN**: Guinea | **GP**: Guadeloupe | **GQ**: Equatorial Guinea | **GR**: Greece | **GT**: Guatemala |
| **GU**: Guam | **GW**: Guinea-Bissau | **GY**: Guyana | **HK**: Hong Kong SAR China | **HN**: Honduras |
| **HR**: Croatia | **HT**: Haiti | **IC**: Canary Islands | **ID**: Indonesia | **IE**: Ireland |
| **IL**: Israel | **IM**: Isle of Man | **IN**: India | **IO**: British Indian Ocean Territory | **IQ**: Iraq |
| **IR**: Iran | **IS**: Iceland | **JE**: Jersey | **JO**: Jordan | **KG**: Kyrgyzstan |
| **KH**: Cambodia | **KI**: Kiribati | **KM**: Comoros | **KN**: St. Kitts & Nevis | **KP**: North Korea |
| **KW**: Kuwait | **KY**: Cayman Islands | **KZ**: Kazakhstan | **LA**: Laos | **LB**: Lebanon |
| **LC**: St. Lucia | **LK**: Sri Lanka | **LR**: Liberia | **LS**: Lesotho | **LT**: Lithuania |
| **LU**: Luxembourg | **LV**: Latvia | **LY**: Libya | **MA**: Morocco | **MC**: Monaco |
| **ME**: Montenegro | **MF**: St. Martin | **MG**: Madagascar | **MH**: Marshall Islands | **MK**: North Macedonia |
| **ML**: Mali | **MM**: Myanmar (Burma) | **MN**: Mongolia | **MO**: Macao SAR China | **MP**: Northern Mariana Islands |
| **MQ**: Martinique | **MR**: Mauritania | **MS**: Montserrat | **MT**: Malta | **MU**: Mauritius |
| **MV**: Maldives | **MW**: Malawi | **MX**: Mexico | **MY**: Malaysia | **MZ**: Mozambique |
| **NA**: Namibia | **NC**: New Caledonia | **NE**: Niger | **NF**: Norfolk Island | **NG**: Nigeria |
| **NI**: Nicaragua | **NR**: Nauru | **NU**: Niue | **NZ**: New Zealand | **OM**: Oman |
| **PA**: Panama | **PE**: Peru | **PF**: French Polynesia | **PG**: Papua New Guinea | **PH**: Philippines |
| **PK**: Pakistan | **PM**: St. Pierre & Miquelon | **PN**: Pitcairn Islands | **PR**: Puerto Rico | **PS**: Palestinian Territories |
| **PT**: Portugal | **PW**: Palau | **QA**: Qatar | **RE**: Réunion | **RU**: Russia |
| **RW**: Rwanda | **SA**: Saudi Arabia | **SB**: Solomon Islands | **SC**: Seychelles | **SD**: Sudan |
| **SE**: Sweden | **SG**: Singapore | **SH**: St. Helena | **SI**: Slovenia | **SJ**: Svalbard & Jan Mayen |
| **SK**: Slovakia | **SL**: Sierra Leone | **SM**: San Marino | **SN**: Senegal | **SO**: Somalia |
| **SR**: Suriname | **SS**: South Sudan | **ST**: São Tomé & Príncipe | **SV**: El Salvador | **SX**: Sint Maarten |
| **SY**: Syria | **SZ**: Eswatini | **TC**: Turks & Caicos Islands | **TD**: Chad | **TG**: Togo |
| **TJ**: Tajikistan | **TK**: Tokelau | **TL**: Timor-Leste | **TM**: Turkmenistan | **TN**: Tunisia |
| **TO**: Tonga | **TT**: Trinidad & Tobago | **TV**: Tuvalu | **TW**: Taiwan | **UA**: Ukraine |
| **UM**: U.S. Outlying Islands | **UY**: Uruguay | **UZ**: Uzbekistan | **VA**: Vatican City | **VC**: St. Vincent & Grenadines |
| **VE**: Venezuela | **VG**: British Virgin Islands | **VI**: U.S. Virgin Islands | **VN**: Vietnam | **VU**: Vanuatu |
| **WF**: Wallis & Futuna | **WS**: Samoa | **XK**: Kosovo | **YE**: Yemen | **YT**: Mayotte |
| **ZM**: Zambia | **ZW**: Zimbabwe |

### How to test the changes in this Pull Request:

1. Run `npm run docker:up` to prepare the testing environment
2. Run `npm run test:e2e` to run the E2E tests 

**Manual Test:**
1. While onboarding, select `Aruba` as your country, and finish onboarding. If you have already finished onboarding, you can go to **WooCommerce > Orders**, then on the above tab drawer, choose **Help > Setup Wizard**, then click on the **Setup Wizard** button to re-access the onboarding workflow.
2. After you finish onboarding, go to `WooCommerce > Settings > General` and on the Currency select, you should see `Aruban florin (Afl.)` as the selected currency for that store.

### Other information:

**Note** The tests will be added separately on this PR: https://github.com/woocommerce/woocommerce/pull/30350

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

Enhancement - Update `locale-info.php` to contain all the countries, currency specs and locale information. #30216

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -8,6 +8,7 @@
 * Enhancement - [Transparency] CLI command for viewing tracking data for your store. #30010
 * Enhancement - All settings pages can now be extended consistently with new sections and settings. Also, unit tests have been added. #27684
 * Enhancement - Set checkout fields value with the default defined value where form is not presented to the user. #29820
+* Enhancement - Update `locale-info.php` to contain all the countries, currency specs and locale information. #30216
```
Comment (roykho) [inline changelog.txt:11]: Can you please remove this change as we usually add the changelogs from our side.

File: i18n/currency-info.php [hunk 2]
```diff
@@ -0,0 +1,952 @@
+<?php
+/**
+ * Currency formatting information
+ *
+ * @package WooCommerce\i18n
+ * @version 5.5.0
```
Comment (roykho) [inline i18n/currency-info.php:6]: Can you please change the version to 5.6.0

File: i18n/locale-info.php [hunk 3]
```diff
@@ -3,326 +3,3980 @@
  * Locales information
  *
  * @package WooCommerce\i18n
- * @version 3.5.0
+ * @version 5.5.0
```
Comment (roykho) [inline i18n/locale-info.php:6]: Can you please change the version to 5.6.0

File: i18n/locale-info.php [hunk 4]
```diff
@@ -3,326 +3,3980 @@
...
+		'name'           => 'Zambian kwacha',
+		'singular'       => 'Zambian kwacha',
+		'plural'         => 'Zambian kwachas',
+		'short_symbol'   => 'ZK',
+		'locales'        => $locales['ZMW'],
 	),
-	'ZA' => array(
-		'currency_code'  => 'ZAR',
+	'ZW' => array(
+		'currency_code'  => 'USD',
```
Comment (roykho) [inline i18n/locale-info.php:3967]: Is this normal to have non US countries to use USD? I see this in other areas as well.

**Review by roykho — CHANGES_REQUESTED** [review]
Thanks for the PR. Tested well per your instructions. I've left some comments which needs to be addressed and also a question.

File: i18n/locale-info.php (hunk 4 above)
Comment (tpaksu) [reply i18n/locale-info.php:3967]: Yes, that surprised me too, but it seems some countries use directly USD. I've checked it from this list of countries: https://www.localeplanet.com/icu/iso3166.html and it seems that way.

Comment (roykho) [reply i18n/locale-info.php:3967]: Hmmm I wonder if that information is correct.  If you look it up, they use Zimbabwean dollars. While they may allow USD currency, their official country currency is Zimbabwean dollars. This is quite confusing... :(

File: changelog.txt (hunk 1 above)
Comment (tpaksu) [reply changelog.txt]: Done.

File: i18n/currency-info.php (hunk 2 above)
Comment (tpaksu) [reply i18n/currency-info.php]: Done.

File: i18n/locale-info.php (hunk 3 above)
Comment (tpaksu) [reply i18n/locale-info.php]: Done.

File: i18n/locale-info.php (hunk 4 above)
Comment (tpaksu) [reply i18n/locale-info.php:3967]: This is the currency history of Zimbabwe according to the CLDR repository (trunk) https://github.com/unicode-org/cldr-json/blob/39205fdc027c935e52a268afcf01cd240f3c442d/cldr-json/cldr-core/supplemental/currencyData.json#L3468-L3498, the values with a `_to` key shows deprecated currencies, without a `_to` key shows the current ones.

Comment (roykho) [reply i18n/locale-info.php:3967]: Yeah I saw that hence it is confusing as other documents state something else. However in the interest of this, we're following CLDR so we can only go by what they state. So I am ok with this change.

**Review by roykho — APPROVED** [review]
LGTM! Thanks for the changes.

**Comment by tpaksu** [comment]
@roykho thanks for the approval and merge!

**Pull request merged**